tests/test_builders/test_build_linkcheck.py

```diff
@@ -965,6 +965,24 @@ def test_TooManyRedirects_on_HEAD(app, monkeypatch):
     }
 
 
+@pytest.mark.sphinx('linkcheck', testroot='linkcheck-localserver')
+def test_ignore_redirection(app):
+    with serve_application(app, InfiniteRedirectOnHeadHandler) as address:
+        app.config.linkcheck_ignore = [f'http://{address}/redirected']
+        app.build()
+
+    with open(app.outdir / 'output.json', encoding='utf-8') as fp:
+        content = json.load(fp)
+    assert content == {
+        'code': 302,
+        'status': 'ignored',
+        'filename': 'index.rst',
+        'lineno': 1,
+        'uri': f'http://{address}/',
+        'info': '/redirected',
```

Comment:
This `info` output doesn't seem particularly useful; it's the contents of the `Location` header from the redirect that was ignored.  Let's try to provide some more-meaningful messaging here.

Reply:
For consistency with existing `_Status.REDIRECTED` results (note: these are only written to the JSON output file, not the txt output file), I think we should provide the source URL as the `uri` field, and the fully-qualified ultimate redirection URL (which in this case will be the one that we stopped at / decided to ignore).